models/DataObject/ClassDefinition/Data/QuantityValue.php

```diff
@@ -257,8 +257,9 @@ public function getDataForResource($data, $object = null, $params = [])
      */
     public function getDataFromResource($data, $object = null, $params = [])
     {
-        if ($data[$this->getName() . '__value'] || $data[$this->getName() . '__unit']) {
-            $quantityValue = new Model\DataObject\Data\QuantityValue((float)$data[$this->getName() . '__value'], $data[$this->getName() . '__unit']);
+        if (strlen($data[$this->getName() . '__value']) > 0 || $data[$this->getName() . '__unit']) {
```

Comment:
Why not simply compare it with  === null instead of the strlen magic ?

Reply:
I tried to be consistent with other existing code/ checks (e.g. ```ClassDefinition\Data\Numeric```). Feel free to modify my PR. If the value is guaranteed to be null in that case (and not an empty string) I'd prefer a type safe comparison as well.

Reply:
I don't see that "Numeric" does anything in getDataFromResource ...

Reply:
Besides that, qv_value is stored as a float in the db ...

Reply:
You're right, it's ```getDataFromResource()``` scope. In that case let's change it to null comparison. Would you mind altering my PR?